Source/Thunder/IRemoteInstantiation.h

```diff
@@ -57,7 +57,7 @@ namespace PluginHost {
             const uint8_t threads,
             const int8_t priority,
             const string configuration,
-            const string environments) = 0;
+            const std::vector<string>& environments) = 0;
```

Comment:
As this is a real COMRPC interface (used for "off host" instantiation, we must make this interface simialr to this: https://github.com/rdkcentral/ThunderInterfaces/blob/master/interfaces/IWifiControl.h#L61-L68
So suggest to add a struct to RPC::Object class (within that scope) like:
`
enum Type : uint8_t {
    LOCAL = 0,
    GLOBAL = 1
};

struct Environment {
    string key;
    string value;
    Type type;
}`

and than here you could state:
`
using IEnvironmentIterator = RPC::IIteratorType<RPC::Object::Environment, ID_ENVIRONMENT_ITERATOR>;

virtual uint32_t Instantiate(
              ...
              IEnvironmentIterator*& environments /* @out */);
`

Realizing this, I also suggest to only do the substitution of the %...% only after this point so that the "host" reciever will fill in its own substitutions. So suggest to do the substitute here: https://github.com/rdkcentral/Thunder/blob/master/Source/Thunder/PluginServer.h#L2416

Reply:
done